packages/flutter/lib/src/painting/image_provider.dart

```diff
@@ -270,10 +271,23 @@ abstract class ImageProvider<T> {
   /// This is the public entry-point of the [ImageProvider] class hierarchy.
   ///
   /// Subclasses should implement [obtainKey] and [load], which are used by this
-  /// method.
+  /// method. If they need to manage the actual resolution of the image, they
+  /// should override [resolveForStream] instead.
+  @nonVirtual
```

Comment:
This is a bigger break that I need to make the way I'm doing `DeferringImageProvider` work. I don't to call `resolve` from there and end up decoding work before I'm ready.

Reply:
Interestingly, this would not break any internal customers or existing customer tests - it only breaks our own test implementations.